### ISSUE TITLE
Set maximum device problem

### ISSUE THREAD

**Issue by rhenku** [vitalidze/traccar-web#215]
i have small problem, if i set user as manager and set maximum device it work, but if user as manager add user and set maximum devices he can set any number maximum devices.. :D so he can add user and give max number he like..

sory my english bad.. :)

**Comment by vitalidze** [comment]
This is a known issue and a subject to change in scope of #214.

**Comment by wolfbgn** [comment]
I dont know if this is the place to mention this. Sorry if I am wrong.
At the moment USER is able to change maximum number of devices and expiration date.
It has to be USER with limited rights not to be able to change these values.
I dont know if I am right but I think that USERS should not be able to change those values.

**Comment by vitalidze** [comment]
Yeah, I know about this issue. It should be fixed in next version along with #214

**Comment by wolfbgn** [comment]
Thank you.

**Comment by draganjovanovich** [comment]
i fixed that by adding this code in userDialog.java and disable ui:field "*****" in xml, i dont know is that correct solution,but it seems to work fine.,. if you want manager to have acces also , just add manager in if statement. 

``` javascript
        if (ApplicationContext.getInstance().getUser().getAdmin()) {
            admin.setEnabled(true);
            manager.setEnabled(true);
            readOnly.setEnabled(true);
            expirationDate.setEnabled(true);
            maxNumOfDevices.setEnabled(true);
        }
        else {
            manager.setEnabled(false);
            admin.setEnabled(false);
            readOnly.setEnabled(false);
            expirationDate.setEnabled(false);
            maxNumOfDevices.setEnabled(false);
        } 
```

**Comment by vitalidze** [comment]
This fixes it only in UI. You **must** restrict it on the server side as well (i.e. in `DataServiceImpl` class).

**Comment by draganjovanovich** [comment]
Cool, i will do that...

**Comment by vitalidze** [comment]
You can then submit a pull request to share your solution.

**Comment by draganjovanovich** [comment]
Ok, tomorrow i will try to change, test it and build.

**Comment by vitalidze** [comment]
Sounds good. Thank you.

**Comment by draganjovanovich** [comment]
How to push changes from intellij ?

**Comment by vitalidze** [comment]
There should be a 'Push changes' item somewhere in 'Git' context menu. You can try searching it in the global search (opened after hitting 'shift shift' (i.e. shift for two times)).

**Comment by draganjovanovich** [comment]
There is a push ,but it wont push it comes with access denied error.

**Comment by vitalidze** [comment]
If you have cloned my repository then it won't allow you to push since you don't have access here. 

The correct workflow is:

1) fork my repository
2) make a push there
3) create a pull request, so I can merge changes from your repository to mine.

Maybe it looks too complex, but this is the modern way to share your changes on github. I believe it's worth to learn this practice. Anyway, as an option you can just send me changed files by email and I will take it to my repository.

**Comment by draganjovanovich** [comment]
Now ?

**Comment by vitalidze** [comment]
It's perfect. Thank you. I will walk through your code and ask you questions directly in #252

**Comment by vitalidze** [comment]
@wolfbgn now user cannot change neither his expiration date not his maximum number of devices, is is included in [0.8.1rc2](https://github.com/vitalidze/traccar-web/releases/tag/0.8.1rc2)

**Comment by vitalidze** [comment]
Implemented, written [news](http://traccar.litvak.su/2015/09/30/big-bugfix-release.html), updated [latest](http://traccar.litvak.su/installation.html) and [demo](http://traccar.litvak.su/demo.html) versions.
